SDC/seed.js

```diff
@@ -0,0 +1,74 @@
+const faker = require( 'faker' );
+// const database = require( './index.js' );
+const Review = require( './Review.js' );
+const fs = require( 'fs' );
+const path = require( 'path' );
+const os = require( 'os' );
+const filename = path.join( __dirname, 'seedable.csv' );
+const writer = fs.createWriteStream( filename );
+writer.write('product_id,user,score,title,body,recommend,date,response_id,ease,value,quality,appearance,works');
+
+const generateScore = () => Math.ceil( Math.random() * 5 );
+
+const generateReviewScores = () => {
+  const subset = [ 'ease', 'value', 'quality', 'appearance', 'works' ];
+  const subsetScores = {
+    ease: generateScore(),
+    value: generateScore(),
+    quality: generateScore(),
+    appearance: generateScore(),
+    works: generateScore(),
+  };
+
+  if (Math.random() <= 0.75) {
+    const index = Math.ceil(Math.random() * subset.length);
+    const subsetName = subset[index];
+    subsetScores[subsetName] = 0;
+  }
+  return subsetScores;
+};
+
+const generateSeedReviewData = () => {
+  const data = [];
+  for (let i = 0; i < 10000; i += 1) { // creates chunks of data to be pushed to CSV
+    // randomly generate between 15 and 25 reviews per product
+    const numberOfReviews = Math.floor(Math.random() * 10) + 15;
+
+    for (let j = 0; j < numberOfReviews; j += 1) {
+      let review = faker.lorem.paragraph();
+      if (review.length > 255) {
+        review = review.slice(0, 255);
+      }
+      const subsetScores = generateReviewScores();
+      const params = {
+        product_id: i + 1,
```

Comment:
If this generateSeedReviewData() function is invoked 5 times, and each time it is invoked, you push 15-25 reviews to each product, you should end up with 10,000 items, each of which have 75-225 reviews. This would only be 750,000-2,250,000 total entries, and I also think that having that many reviews for a relatively small number of reviews is kind of 'lumpy' data. Do you have a plan to get up to 10,000,000 total pieces of data? I'd be interested to hear what your thoughts are on 'spreading out' your data so you have more items with fewer reviews each.